public/app/plugins/panel/state-timeline/utils.ts

```diff
@@ -350,16 +352,20 @@ export function mergeThresholdValues(field: Field, theme: GrafanaTheme2): Field
       prev = undefined;
     }
     const active = getActiveThreshold(v, thresholds.steps);
-    if (active === prev) {
-      vals[i] = undefined;
-    } else {
-      vals[i] = thresholdToText.get(active);
-    }
+    vals[i] = thresholdToText.get(active);
     prev = active;
   }
 
   return {
     ...field,
+    config: {
+      ...field.config,
+      custom: {
+        ...field.config.custom,
+        // magic value for join() to leave nulls alone
+        spanNulls: -1,
```

Comment:
can the user override this in the panel?

Reply:
we have not exposed control over it in this panel -- it's blocked on a code re-org that we need to do soon. (see https://github.com/grafana/grafana/pull/36622#issuecomment-1016885766)